paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -176,11 +173,11 @@ def get_services_to_k8s_namespaces(
 def sync_all_secrets(
     kube_client: KubeClient,
     cluster: str,
-    services_to_k8s_namespaces: Mapping[str, Set],
+    services_to_k8s_namespaces: Dict[str, Set],
     secret_provider_name: str,
-    vault_cluster_config: Mapping[str, str],
+    vault_cluster_config: Dict[str, str],
     soa_dir: str,
-    vault_token_file: str = DEFAULT_VAULT_TOKEN_FILE,
+    vault_token_file: Optional[str] = None,
```

Comment:
just curious: why did we need to change the default value?

Reply:
and will the callees of this function handle potentially getting a `None` value? I might be missing something, but I doesn't look like we've added any error handling for the `None` case

Reply:
When testing on a playground, some commands were not propagating vault-token because it was being set as null. So default values actually serve no purpose if there are cases when `None` is being passed.
I had to add the following
https://github.com/Yelp/paasta/blob/b9d5953d8c9613ad23a06064494f185f5b2d1146/paasta_tools/cli/cmds/secret.py#L421-L424
to pass the token, and the default argument actually confused me as to why the default argument was not being used
https://github.com/Yelp/paasta/blob/b9d5953d8c9613ad23a06064494f185f5b2d1146/Makefile#L164

Reply:
Let's change when making changes to `vault.py`